### PR TITLE
HW#8: Routing module

### PR DESCRIPTION
**Pull request by mariia-soshnikova-EPAM**
### Homework summary

In scope of this homework I've added routes to my application, getting app state from URL and updating URL when changes made to state.

---
Added functionality:

1.  Now when the address bar pathname is changed to `/`, the user is automatically redirected to `/search`
2. When the address bar pathname is changed to `/search` without a `:searchQuery` URL segment, the search page with the most popular movies is displayed (by sending a request with empty "search" query param), search input value remains empty
3. When the address bar pathname is changed to `/search/:searchQuery` where `:searchQuery` is not an empty string, the search page with results is displayed and the search results match `:searchQuery` from the URL segment, search input value is equal to `:searchQuery`

----
- [x] AC1 AC2 are met
- [x] AC1 AC2 AC3 AC4 are met
- [x] AC1 AC2 AC3 AC4 AC6 AC7 AC9 are met
- [ ] AC1 AC2 AC3 AC4 **AC5** AC6 AC7 **AC8** AC9 AC10 AC11 are met

### CODE REVIEW

File: NetflixRouletteWebpack/src/components/error-page/ErrorPage.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { useRouteError } from "react-router-dom";
+import "./ErrorPage.scss";
+
+export default function ErrorPage() {
+  const error = useRouteError();
+  console.error(error);
+  console.log(error);
+  console.log("hey");
```
Comment (NikitaUstsiushenka) [inline NetflixRouletteWebpack/src/components/error-page/ErrorPage.js]: @Marysosh Do we really need to log "hey"?

File: NetflixRouletteWebpack/src/components/modals/edit-movie-modal/EditMovieModal.js [hunk 2]
```diff
@@ -63,7 +63,11 @@ function EditMovieModal({
       }
     },
     onReset: () => {
-      setSelectedGenres([]);
+      if (modalTitle === "Add movie") {
```
Comment (NikitaUstsiushenka) [inline NetflixRouletteWebpack/src/components/modals/edit-movie-modal/EditMovieModal.js]: @Marysosh Can be simplified like `setSelectedGenres(modalTitle === "Add movie" ? "" : initialValues.genre);`

File: NetflixRouletteWebpack/src/index.js [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+  {
+    path: "search",
+    element: <App />,
+    errorElement: <ErrorPage />,
+    children: [
+      {
+        path: ":searchQuery",
+        element: <App />,
+        loader: ({ params }) => {
+          // const genreArray =
```
Comment (NikitaUstsiushenka) [inline NetflixRouletteWebpack/src/index.js]: @Marysosh Please remove extra comments

File: NetflixRouletteWebpack/src/components/error-page/ErrorPage.js (hunk 1 above)
Comment (mariia-soshnikova-EPAM) [reply NetflixRouletteWebpack/src/components/error-page/ErrorPage.js]: Of course we shouldn't😌 Removed this

File: NetflixRouletteWebpack/src/components/modals/edit-movie-modal/EditMovieModal.js (hunk 2 above)
Comment (mariia-soshnikova-EPAM) [reply NetflixRouletteWebpack/src/components/modals/edit-movie-modal/EditMovieModal.js]: Yes, you're right, it looks better. Corrected✅

File: NetflixRouletteWebpack/src/index.js (hunk 3 above)
Comment (mariia-soshnikova-EPAM) [reply NetflixRouletteWebpack/src/index.js]: Done✅

**Review by NikitaUstsiushenka — APPROVED** [review]

**Pull request merged**